src/gateway/runGateway.ts

```diff
@@ -70,7 +70,7 @@ export async function initGatewayDb(db: Knex) {
  * TX_CONFIRMATION_SUCCESSFUL_ROUNDS query rounds (to randomly selected at each round peers).
```

Comment:
Sorry if it's a silly idea but have you consider deciding on the transaction confirmation status by checking the confirmation status of the block (hash) in which the transaction was included. Potential benefits:
1) Fewer roundtrips for multiple smart contract transaction in single block
2) Possible faster answers about block status as it's more widely used parameter crucial for building new blocks

Reply:
All the orphaned transactions for our loot contract come from one block - https://d1o5nlqr4okus2.cloudfront.net/gateway/interactions?contractId=Daj-MNSnH55TDfxqC7v4eq0lKzVIwh98srUaWqyuZtY&confirmationStatus=orphaned , blockId `bAUsEgzlbGyhJpyg-klaqw9GXhrqiVpbHsV2tjpGJ02faWGzLtuisqGagTqk5tQS`

Yet this block is normally visible on Arweave - https://arweave.net/block/hash/bAUsEgzlbGyhJpyg-klaqw9GXhrqiVpbHsV2tjpGJ02faWGzLtuisqGagTqk5tQS
and viewblock - https://viewblock.io/arweave/block/769686

How exactly this block-based verification should work?

Reply:
I've made some more research...it seems that the "orphaned" transactions issue does not come from the forked blocks...it looks like the arweave gateway caches the transactions that are gossiped for the next block, but does not verify if they are finally becoming part of this block (?)

From the above example:
 1. block height `769686`:
 - indep_hash `bAUsEgzlbGyhJpyg-klaqw9GXhrqiVpbHsV2tjpGJ02faWGzLtuisqGagTqk5tQS`, - https://arweave.net/block/hash/bAUsEgzlbGyhJpyg-klaqw9GXhrqiVpbHsV2tjpGJ02faWGzLtuisqGagTqk5tQS - https://viewblock.io/arweave/block/769686
 
 2. block height `769687`:
 - indep_hash `1SWZh9jw23JMCn0sfLCCwtZ7L9Xb3wkSWJO7MbXy7FLieyyjkz14LcnDEWobvu31`, 
 - previous_block `bAUsEgzlbGyhJpyg-klaqw9GXhrqiVpbHsV2tjpGJ02faWGzLtuisqGagTqk5tQS`
 - https://arweave.net/block/hash/1SWZh9jw23JMCn0sfLCCwtZ7L9Xb3wkSWJO7MbXy7FLieyyjkz14LcnDEWobvu31
 - https://viewblock.io/arweave/block/769687
 
 3. block height `769688`:
 - indep_hash `-QZNOuZyQj27rdGT3SfDp2I8xayG81eyLd7bGtv03xHiBfbu_v93BXUTfwxgyoCr`,
 - previous_block `1SWZh9jw23JMCn0sfLCCwtZ7L9Xb3wkSWJO7MbXy7FLieyyjkz14LcnDEWobvu31`
 - https://arweave.net/block/hash/-QZNOuZyQj27rdGT3SfDp2I8xayG81eyLd7bGtv03xHiBfbu_v93BXUTfwxgyoCr
 - https://viewblock.io/arweave/block/769688
 
 4. block height `769689`:
 - indep_hash `7s2pOt_0wAWfK3x4Tqk2Mro0DEpRwwquTL59BeW_o3gQzrsB3-ZF8RcfpVaWu5qs`
 - previous_block `-QZNOuZyQj27rdGT3SfDp2I8xayG81eyLd7bGtv03xHiBfbu_v93BXUTfwxgyoCr`
 - https://arweave.net/block/hash/7s2pOt_0wAWfK3x4Tqk2Mro0DEpRwwquTL59BeW_o3gQzrsB3-ZF8RcfpVaWu5qs
 - https://viewblock.io/arweave/block/769689
 
 5. block height `769690`:
 - indep_hash `mj52-sl7iOdNFFkkxqyQ9WTPjxwmy4I6rp3ZW81wKKIgdpwTqCh79g1a0N8qYe17`
 - previous_block `7s2pOt_0wAWfK3x4Tqk2Mro0DEpRwwquTL59BeW_o3gQzrsB3-ZF8RcfpVaWu5qs`
 - https://arweave.net/block/hash/mj52-sl7iOdNFFkkxqyQ9WTPjxwmy4I6rp3ZW81wKKIgdpwTqCh79g1a0N8qYe17
 - https://viewblock.io/arweave/block/769690
 
 ...and so on...TBH I don't think that all of these blocks are part of the fork...
 
 Though there's probably a bit more efficient way to verify transactions that have just come to my mind:
 1. load a bunch of `non_confimed` interactions and group them by block indep_hash
 2. load blocks data for all the blocks from point 1 (using the `https://arweave.net/block/hash/<indep_hash>` endpoint
 3. verify if the transactions are within the `txs` field from the response from point 2...
 
In case of our loot contract - none of the orphaned transactions are on the `txs` list of the `bAUsEgzlbGyhJpyg-klaqw9GXhrqiVpbHsV2tjpGJ02faWGzLtuisqGagTqk5tQS`.
 
![image](https://user-images.githubusercontent.com/84311757/145973374-486fa490-02c8-4a11-b544-3a46b5bdc8a0.png)

This could potentially generate much less requests...

To sum up - I believe that "orhpaned" in this case means "transactions that are not within any block (but for some reason were cached by the gateway" and does NOT mean "transactions that are part of a forked block".